rollup/fees/rollup_fee.go

```diff
@@ -48,59 +45,87 @@ type StateDB interface {
 	GetBalance(addr common.Address) *big.Int
 }
 
-// CalculateL1MsgFee computes the L1 portion of the fee given
-// a Message and a StateDB
-// Reference: https://github.com/ethereum-optimism/optimism/blob/develop/l2geth/rollup/fees/rollup_fee.go
-func CalculateL1MsgFee(msg Message, state StateDB) (*big.Int, error) {
-	tx := asTransaction(msg)
+func EstimateL1DataFeeForMessage(msg Message, baseFee, chainID *big.Int, signer types.Signer, state StateDB) (*big.Int, error) {
+	if msg.IsL1MessageTx() {
+		return big.NewInt(0), nil
+	}
+
+	unsigned := asUnsignedTx(msg, baseFee, chainID)
+	// with v=1
```

Comment:
Why `v = 0x01` and not `v = 0xff` or `v = 0xffffff`?

Reply:
v will only be 0 or 1, see core/types/transaction_signing.go

Using 1 to estimate seems more accurate

Reply:
I see. The chainId is added to `v` (for legacy tx) in the `WithSignature` call.